wherehows-web/app/constants/dataset-compliance.ts

```diff
@@ -264,6 +269,81 @@ const createInitialComplianceInfo = (datasetId: string): IComplianceInfo => {
   };
 };
 
+/**
+ * Extracts the values on a compliance Entity for a given list of keys
+ * @template K IComplianceEntity instance attribute
+ * @param {Array<K>} [keys=[]]
+ * @param {string} fieldName
+ * @param {IComplianceInfo.complianceEntities} [source=[]]
+ * @returns {({ [V in K]: IComplianceEntity[V] } | {})}
+ */
+const getKeysOnComplianceEntity = <K extends keyof IComplianceEntity>(
+  keys: Array<K> = [],
+  fieldName: string,
+  source: IComplianceInfo['complianceEntities'] = []
+): { [V in K]: IComplianceEntity[V] } | {} => {
+  const sourceField: IComplianceEntity | void = source.find(({ identifierField }) => identifierField === fieldName);
+  let result = {};
+
+  if (sourceField) {
+    for (const [key, value] of <Array<[K, IComplianceEntity[K]]>>Object.entries(sourceField)) {
```

Comment:
You can use `Array#reduce` haha

Reply:
haha, i got lazy in refactoring here, i'll update